ios/browser/api/https_upgrades/https_upgrade_service_factory.mm

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/api/https_upgrades/https_upgrade_service_factory.h"
+
+#include "brave/ios/browser/api/https_upgrades/https_upgrade_service+private.h"
+#include "ios/chrome/browser/https_upgrades/model/https_upgrade_service_factory.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+#include "ios/components/security_interstitials/https_only_mode/https_upgrade_service.h"
+
+@implementation BraveHttpsUpgradeServiceFactory
++ (nullable id)serviceForBrowserState:(ChromeBrowserState*)browserState {
```

Comment:
Where is this function used? I don't see it called anywhere?

Reply:
`KeyedServiceFactoryWrapper` implements `get(privateMode:)`, used here: https://github.com/brave/brave-core/pull/25454/files#diff-ad72b6236436552cbcd002fc1d11466c924a08dd2ddeceee578e4b464cb8f36eR64